File: src/lib/tauri-api/agent-chat.ts
import type { AgentChatEvent, AgentChatSession, CreateAgentChatSessionInput, SendAgentChatMessageInput } from '../../types/agent-chat';
import { invokeCommand } from './client';

export function createAgentChatSession(input: CreateAgentChatSessionInput): Promise<AgentChatSession> {
  return invokeCommand<AgentChatSession>('create_agent_chat_session', { input });
}

export function sendAgentChatMessage(input: SendAgentChatMessageInput): Promise<AgentChatEvent> {
  return invokeCommand<AgentChatEvent>('send_agent_chat_message', { input });
}

export function listAgentChatSessions(workspaceId: string): Promise<AgentChatSession[]> {
  return invokeCommand<AgentChatSession[]>('list_agent_chat_sessions', { workspaceId });
}

export function listAgentChatEvents(sessionId: string): Promise<AgentChatEvent[]> {
  return invokeCommand<AgentChatEvent[]>('list_agent_chat_events', { sessionId });
}

export function interruptAgentChatSession(sessionId: string): Promise<AgentChatSession> {
  return invokeCommand<AgentChatSession>('interrupt_agent_chat_session', { sessionId });
}

export function closeAgentChatSession(sessionId: string): Promise<AgentChatSession> {
  return invokeCommand<AgentChatSession>('close_agent_chat_session', { sessionId });
}

File: src-tauri/src/commands/agent_chat.rs
use tauri::State;

use crate::models::{
    AgentChatEvent, AgentChatSession, CreateAgentChatSessionInput, SendAgentChatMessageInput,
};
use crate::services::agent_chat_service;
use crate::state::AppState;

#[tauri::command]
pub fn create_agent_chat_session(
    state: State<'_, AppState>,
    input: CreateAgentChatSessionInput,
) -> Result<AgentChatSession, String> {
    agent_chat_service::create_agent_chat_session(&state, input)
}

#[tauri::command]
pub fn send_agent_chat_message(
    state: State<'_, AppState>,
    input: SendAgentChatMessageInput,
) -> Result<AgentChatEvent, String> {
    agent_chat_service::send_agent_chat_message(&state, input)
}

#[tauri::command]
pub fn list_agent_chat_sessions(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<Vec<AgentChatSession>, String> {
    agent_chat_service::list_agent_chat_sessions(&state, &workspace_id)
}

#[tauri::command]
pub fn list_agent_chat_events(
    state: State<'_, AppState>,
    session_id: String,
) -> Result<Vec<AgentChatEvent>, String> {
    agent_chat_service::list_agent_chat_events(&state, &session_id)
}

#[tauri::command]
pub fn interrupt_agent_chat_session(
    state: State<'_, AppState>,
    session_id: String,
) -> Result<AgentChatSession, String> {
    agent_chat_service::interrupt_agent_chat_session(&state, &session_id)
}

#[tauri::command]
pub fn close_agent_chat_session(
    state: State<'_, AppState>,
    session_id: String,
) -> Result<AgentChatSession, String> {
    agent_chat_service::close_agent_chat_session(&state, &session_id)
}
